src/main/java/org/opensearch/sdk/authz/ProtectedRoute.java

```diff
@@ -0,0 +1,71 @@
+package org.opensearch.sdk.authz;
+
+import org.opensearch.rest.RestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.sdk.ExtensionRestResponse;
+import org.opensearch.sdk.handlers.ExtensionRouteRequestHandler;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ProtectedRoute extends RestHandler.Route {
+
+    private static final Pattern ROUTE_PARAM_REGEX = Pattern.compile("\\{(.*?)\\}");
+
+    public ExtensionRouteRequestHandler requestHandler;
+
+    private String requiredPermission;
+    public ProtectedRoute(RestRequest.Method method, String path, String requiredPermission, ExtensionRouteRequestHandler requestHandler) {
+        super(method, path);
+        this.requestHandler = requestHandler;
+        this.requiredPermission = requiredPermission;
+    }
+
+    public String getRequiredPermission() {
+        return this.requiredPermission;
+    }
+
+    public Pattern getRouteRegex() {
+        if (getPath() == null) {
+            return null;
+        }
+        String routeRegex = getPath().replaceAll("\\{.*?\\}", "(.*)");
```

Comment:
Regexes are sometimes mysterious.  This one isn't obvious what it's doing or why.  It looks like it's to conform to the expectation of `ROUTE_PARAM_REGEX` but I'm not sure why that is, either.

Can we please have a comment explaining this?  (Unless you switch to glob matching per the comment below...)